File: 2094. Finding 3-Digit Even Numbers/findEvenNumbers.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var findEvenNumbers = function(digits) {
    let ans = [];
    let cnt = new Array(10).fill(0);

    for(let num of digits){
        cnt[num]++
    }
    
    for(let i=100; i<=999; i+=2){           // O(n)
        //check if all integers are include in (digits) array
        let three = Math.floor(i / 100)
        let two = Math.floor(i / 10) % 10
        let one = i % 10
        let success = true

        if(--cnt[three] < 0){
            success = false
        }
        if(--cnt[two] < 0){
            success = false
        }
        if(--cnt[one] < 0){
            success = false
        }

        if(success)ans.push(i)

        cnt[three]++
        cnt[two]++
        cnt[one]++
    }

    return ans;
};

File: 2094. Finding 3-Digit Even Numbers/findEvenNumbers.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;


class Solution {
    public:
        vector<int> findEvenNumbers(vector<int>& digits) {
            vector<int> ans;
            vector<int> cnt(10,0);
            int max = 0;
    
            sort(digits.begin(), digits.end(), greater<>());
            for(int i=0 ; i<3; i++){
                max += digits[i];
                if(i<2) max *= 10;
            }
    
            for(auto num : digits){
                cnt[num]++;
            }
    
            for(int i=100; i<=max; i+=2){
                int three = i / 100;
                int two = (i / 10) % 10;
                int one = i % 10;
                bool success = true;
    
                if(--cnt[three] < 0)success = false;
                if(--cnt[two] < 0)success = false;
                if(--cnt[one] < 0)success = false;
    
                if(success)ans.push_back(i);
    
                cnt[three]++;
                cnt[two]++;
                cnt[one]++;
            }
            return ans;
        }
    };